src/main/webapp/dev/js/common/instructor.js

```diff
@@ -358,14 +358,19 @@ function bindCourseDeleteAllLinks() {
         event.preventDefault();
 
         const $clickedLink = $(event.currentTarget);
-        const messageText = 'Are you sure you want to permanently delete all the courses? '
-                + 'This operation will delete all students and sessions in these courses. '
-                + 'All instructors of these courses will not be able to access them hereafter as well.';
+        const $softDeletedCourses = $('td[id^="recoverycourseid"]');
```

Comment:
It's not very nice to couple with the DOM here.

We have two choices:
* Use AJAX to retrieve the actual data from the backend
* Explain the coupling (retrieving data from the table)

Understandably the first option is better but if it requires too much refactoring or too much work it is not worth it.  Up to you to decide :)

For the second option, we don't necessarily have to couple with a `td` element either given that we already have `id`. You can make the option appropriately but please explain. Also, if you are going with this option let's convert it to an array so we can iterate it using ES6 iteration methods.

Reply:
I'm using this to retrieve all courses from Recycle Bin, because I saw the same usage in `instructorCourses.js`.

Reply:
Our code is always evolving so it's up to us make better decisions if we can :)

Not to mention in the past when we were using less AJAX, and DOM manipulation was more popular until the declarative style caught on to make UI simpler to reason about.

Reply:
Yeah there's no need to couple with `td`. I think here we just want to make it clear that we are retrieving data from a column of the table.

Reply:
Well, suppose if we want to make it clear that we are retrieving the data from a table then when we change the UI we need to change the logic here too. What if the data doesn't come in a table?

While that isn't likely to change in this case, separating data and presentation allows for more flexibility in how the data is presented.

Take for example we can easily change the courses to be placed in a list rather than a table. In that case, if the Javascript is coupled to the DOM then when the UI change we have to change the logic as well.